zetaclient/chains/ton/observer/observer.go

```diff
@@ -0,0 +1,80 @@
+package observer
+
+import (
+	"context"
+	"errors"
+
+	"github.com/tonkeeper/tongo/tlb"
+	"github.com/tonkeeper/tongo/ton"
+
+	"github.com/zeta-chain/node/pkg/bg"
+	toncontracts "github.com/zeta-chain/node/pkg/contracts/ton"
+	"github.com/zeta-chain/node/x/crosschain/types"
+	"github.com/zeta-chain/node/zetaclient/chains/base"
+	"github.com/zeta-chain/node/zetaclient/chains/interfaces"
+)
+
+// Observer is a TON observer.
+type Observer struct {
+	*base.Observer
+
+	client  LiteClient
+	gateway *toncontracts.Gateway
+}
+
+// LiteClient represents a TON client
+//
+//go:generate mockery --name LiteClient --filename ton_liteclient.go --case underscore --output ../../../testutils/mocks
+type LiteClient interface {
```

Comment:
We have all interfaces together in `interfaces.go` if you want to put it somewhere else.

Reply:
It's actually a bad practice, I want to move away from this concept of having a centralized place for interfaces.

https://www.perplexity.ai/search/golang-define-interfaces-where-mK1pR.HvRNic4eaDDng_OA